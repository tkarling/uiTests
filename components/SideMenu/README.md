# @ui-tests/side-menu

Welcome to my `SideMenu` component.

## Installation

```sh
npm i @ui-tests/side-menu
# or with yarn
yarn add @ui-tests/side-menu
```

Then to use the component in your code just import it!

```js
import SideMenu from '@ui-tests/side-menu';
// and with css-modules
import '@ui-tests/side-menu/dist/main.css';
```
