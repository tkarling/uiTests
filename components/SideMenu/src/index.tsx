import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import Octicon, { ThreeBars, Person } from '@primer/octicons-react';
import useMedia from './useMedia';

const Container = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  header {
    background-color: steelblue;
    color: aliceblue;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    padding: 10px;
    .user {
      display: flex;
      flex-direction: row;
      .userName {
        margin-left: 10px;
      }
    }
  }
  .contentContainer {
    flex: 1;
    display: flex;
    flex-direction: row;
    nav {
      background-color: lightsteelblue;
      color: steelblue;
      padding: 10px;
      min-width: 150px;
    }
    .content {
      background-color: aliceblue;
      flex: 1;
      display: flex;
      align-items: stretch;
      justify-content: center;
      padding: 10px;
    }
  }
`;

export const SideMenu = ({ children }: { children: Node }) => {
  const { xs } = useMedia();

  const [open, setOpen] = useState(() => !xs);
  const toggle = () => setOpen(o => !o);
  useEffect(() => {
    setOpen(!xs);
  }, [xs]);

  return (
    <Container>
      <header>
        <div role="presentation" onClick={toggle}>
          {xs ? <Octicon icon={ThreeBars} /> : <div>CompanyLogo</div>}
        </div>
        <div className="user">
          <Octicon icon={Person} />
          {!xs && <div className="userName">Tuija Karling</div>}
        </div>
      </header>
      <div className="contentContainer">
        {open && <nav>sidemenu</nav>}
        <div className="content">{children}</div>
      </div>
    </Container>
  );
};

export default SideMenu;
