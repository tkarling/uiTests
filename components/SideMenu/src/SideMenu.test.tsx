import React from 'react';
import { render } from '@testing-library/react';
import SideMenu from '.';

describe('SideMenu', () => {
  test('It matches the snapshot', () => {
    const { container } = render(<SideMenu />);
    expect(container).toMatchSnapshot();
  });
});
