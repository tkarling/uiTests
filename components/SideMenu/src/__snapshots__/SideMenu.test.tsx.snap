// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu It matches the snapshot 1`] = `
<div>
  <div
    class="src__Container-sc-17aznbp-0 iMFenz"
  >
    <header>
      <div
        role="presentation"
      >
        <div>
          CompanyLogo
        </div>
      </div>
      <div
        class="user"
      >
        <svg
          aria-hidden="true"
          class="octicon"
          height="16"
          role="img"
          style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
          viewBox="0 0 12 16"
          width="12"
        >
          <path
            d="M12 14.002a.998.998 0 01-.998.998H1.001A1 1 0 010 13.999V13c0-2.633 4-4 4-4s.229-.409 0-1c-.841-.62-.944-1.59-1-4 .173-2.413 1.867-3 3-3s2.827.586 3 3c-.056 2.41-.159 3.38-1 4-.229.59 0 1 0 1s4 1.367 4 4v1.002z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="userName"
        >
          Tuija Karling
        </div>
      </div>
    </header>
    <div
      class="contentContainer"
    >
      <nav>
        sidemenu
      </nav>
      <div
        class="content"
      />
    </div>
  </div>
</div>
`;
